### PR TITLE
Modify query show loading icon instead of 0 studies

### PR DESCRIPTION
**Pull request by dippindots**
improve loading behavior of query selector when default selection present.
Fix # https://github.com/cbioportal/cbioportal/issues/5911.

### CODE REVIEW

File: src/shared/components/query/CancerStudySelector.tsx [hunk 1]
```diff
@@ -173,90 +173,92 @@ export default class CancerStudySelector extends React.Component<ICancerStudySel
...
-                        )}
-
-
-                        {(!!(!this.store.forDownloadTab) && !!(!this.store.cancerTypes.isPending && !this.store.cancerStudies.isPending)) && (
-                            <Observer>
-                                {() => {
-                                    let hasSelection = this.store.selectableSelectedStudyIds.length > 0;
-
-                                    if (hasSelection) {
+                    {this.store.selectableSelectedStudyIds.length >= 1 && (
```
Comment (alisman) [inline src/shared/components/query/CancerStudySelector.tsx]: this PR show a lot of change, but i think only line 176 is changed, right?  can you confirm that?

**Comment by alisman** [comment]
This PR hides the study selection summary, but in fact, the whole query form is innaccurate because it doesn't show the selected studies until the studies endpoint returns.  So, it probably makes more sense to just not show the form at (and show a loader) until we have the data necessary to show selection.

File: src/shared/components/query/CancerStudySelector.tsx (hunk 1 above)
Comment (dippindots) [reply src/shared/components/query/CancerStudySelector.tsx]: @alisman Yes, just line 176 is changed.

**Comment by dippindots** [comment]
@alisman `selectableSelectedStudyIds` is an observable value, not a promise. I am not sure how could we know `selectableSelectedStudyIds` has been successfully updated.

**Comment by dippindots** [comment]
@alisman changed to wait the `selectableStudiesSet` completed, then render the component. Can use Ino's [link](https://cbioportal-frontend-pr-2215.herokuapp.com/results/mutations?case_set_id=all&gene_list=EGFR&cancer_study_list=5c8a7d55e4b046111fee2296) to test.

**Comment by dippindots** [comment]
@alisman In the file `QueryContainer.tsx`, just L82, L138-L140 has been changed.

File: src/shared/components/query/QueryContainer.tsx [hunk 2]
```diff
@@ -77,61 +78,66 @@ export default class QueryContainer extends React.Component<QueryContainerProps,
...
-						</span>
-						)}
-
-						{this.store.oqlMessages.map(msg=>{
-							return (
-								<span className={styles.oqlMessage}>
-									<i className='fa fa-info-circle' style={{marginRight: 5}}/>
-									{msg}
+				{   
+					(!this.store.defaultSelectedIds.size || this.store.defaultSelectedIds.size && this.store.selectableSelectedStudyIds.length > 0 && this.store.initiallySelected.profileIds && this.store.cancerTypes.isComplete && this.store.cancerStudies.isComplete && this.store.profiledSamplesCount.isComplete) ? (
```
Comment (alisman) [inline src/shared/components/query/QueryContainer.tsx]: clean this up according to conversation

Comment (alisman) [reply src/shared/components/query/QueryContainer.tsx]: put this in a method

**Comment by alisman** [comment]
Also, just noticed that layout of query UI is slightly messed up
image
See spacing between regions is collapsed

**Comment by dippindots** [comment]
@alisman Updated and rebased to latest master, pull the conditions out to `defaultStudiesInitialized()` and use `react-if` to avoid the layout change.

**Comment by alisman** [comment]
@dippindots sorry, but we need to squash these two commit and give them a better commit message.  like,  "improve loading behavior of query selector when default selection present"

**Pull request merged**